math/src/polynom/multi_linear.rs

```diff
@@ -0,0 +1,240 @@
+        inner_product(self.evaluations.iter().copied(), lagrange_kernel.iter().copied())
+    }
+
+    /// Computes f(r_0, y_1, ..., y_{ν - 1}) using the linear interpolation formula
+    /// (1 - r_0) * f(0, y_1, ..., y_{ν - 1}) + r_0 * f(1, y_1, ..., y_{ν - 1}) and assigns
+    /// the resulting multi-linear, defined over a domain of half the size, to `self`.
+    pub fn bind_least_significant_variable(&mut self, round_challenge: E) {
+        let mut result = vec![E::ZERO; 1 << (self.num_variables() - 1)];
+        for (i, res) in result.iter_mut().enumerate() {
+            *res = self.evaluations[i << 1]
+                + round_challenge * (self.evaluations[(i << 1) + 1] - self.evaluations[i << 1]);
+        }
+        *self = Self::from_evaluations(result)
+            .expect("should not fail given that it is a multi-linear");
+    }
+
+    /// Given the multilinear polynomial f(y_0, y_1, ..., y_{ν - 1}), returns two polynomials:
+    /// f(0, y_1, ..., y_{ν - 1}) and f(1, y_1, ..., y_{ν - 1}).
+    pub fn project_least_significant_variable(&self) -> (Self, Self) {
+        let mut p0 = Vec::with_capacity(self.num_evaluations() / 2);
+        let mut p1 = Vec::with_capacity(self.num_evaluations() / 2);
+        for chunk in self.evaluations.chunks_exact(2) {
+            p0.push(chunk[0]);
+            p1.push(chunk[1]);
+        }
+
+        (
+            MultiLinearPoly::from_evaluations(p0).unwrap(),
+            MultiLinearPoly::from_evaluations(p1).unwrap(),
+        )
+    }
+}
+
+impl<E: FieldElement> Index<usize> for MultiLinearPoly<E> {
+    type Output = E;
+
+    fn index(&self, index: usize) -> &E {
+        &(self.evaluations[index])
+    }
+}
+
+impl<E> Serializable for MultiLinearPoly<E>
+where
+    E: FieldElement,
+{
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let Self { num_variables, evaluations } = self;
+        num_variables.write_into(target);
+        evaluations.write_into(target);
+    }
+}
+
+impl<E> Deserializable for MultiLinearPoly<E>
+where
+    E: FieldElement,
+{
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        Ok(Self {
+            num_variables: Deserializable::read_from(source)?,
+            evaluations: Deserializable::read_from(source)?,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum MultiLinearPolyError {
+    #[error("A multi-linear polynomial should have a power of 2 number of evaluations over the Boolean hyper-cube")]
+    EvaluationsNotPowerOfTwo,
+}
+
+// EQ FUNCTION
+// ================================================================================================
+
+/// The EQ (equality) function is the binary function defined by
+///
+/// ```ignore
+/// EQ:    {0 , 1}^ν ⛌ {0 , 1}^ν ⇾ {0 , 1}
+///   ((x_0, ..., x_{ν - 1}), (y_0, ..., y_{ν - 1})) ↦ \prod_{i = 0}^{ν - 1} (x_i * y_i + (1 - x_i)
+/// * (1 - y_i))
+/// ```
```

Comment:
These, and other formulas, we should explicitly write as formals (i.e., using `$$` in comments) so that they are displayed nicely in the docs.

Reply:
Added but not sure if it is the right way to do it. Do you have some documentation to help?